sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3421,6 +3428,14 @@ object functions {
   /**
    * Returns an array of elements after applying a transformation to each element
    * in the input array.
+   * {{{
+   *   df.select(transform(col("i"), (x, i) => x + i))
+   *   df.selectExpr("transform(i, (x, i) -> x + i))")
+   * }}}
+   *
+   * @param column the input array column
+   * @param f: (col, index) => transformed_col the lambda function to filter the input column
+   *           given the index. Indices start at 0.
```

Comment:
have you run the example and verify that the indices start at 0?

Reply:
Examples are copied from tests and it starts from 0 if you look at the test result:

https://github.com/apache/spark/blob/8aebc80e0e67bcb1aa300b8c8b1a209159237632/sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala#L2026